### ISSUE TITLE
Issue deploying to Azure Gov

### ISSUE THREAD

**Issue by Tukeyz** [Azure/missionlz#500]
I have no idea as i dont know coding, but i am getting this error 

"[parameters('hubVirtualNetworkDiagnosticsMetrics')]"
The template has specified this parameter as type 'array', but the default value is specified as type 'string'. This template may not deploy correctly.
**Unable to parse value to type 'array'. The value must be a valid JSON array.**
how do i resolve this error message

**Comment by techdawg70** [comment]
Hi Brooke - this error is seen when a user navigates the github repo to the Command-Line-Deployment section then clicks the "Deploy" link under the "Deploy with Bicep (recommended)" option block. From here the user clicks the button "Deploy to Azure Gov" which redirects the user to their Az Gov tenant -> custom deployment template for MLZ.  This error is seen against the following arears -> Identity Virtual Network Diagnostics Logs, Identity Virtual Network Diagnostics Metrics, Identity Network Security Group Rules, Identity Network Security Group Diagnostics Logs, Identity Network Security Group Diagnostics Metrics, Identity Subnet Service Endpoints.

**Comment by techdawg70** [comment]
Sorry didn't scroll down all the way so I missed additional sections of the template which this error. Those are Operations Virtual Network Diagnostics Logs, Operations Virtual Network Diagnostics Metrics, Operations Network Security Group Rules, Operations Network Security Group Diagnostics Logs, Operations Network Security Group Diagnostics Metrics, Operations Subnet Service Endpoints, Shared Services Virtual Network Diagnostics Logs, Shared Services Virtual Network Diagnostics Metrics, Shared Services Network Security Group Rules, Shared Services Network Security Group Diagnostics Logs, Shared Services Network Security Group Diagnostics Metrics, Shared Services Subnet Service Endpoints

**Comment by brooke-hamilton** [comment]
Thanks for the additional info. I will take a look. 💯

**Comment by glennmusa** [comment]
It appears the custom deployment template doesn't support parsing shared array parameters:

```
The template has specified this parameter as type 'array', but the default value is specified as type 'string'. This template may not deploy correctly.
```

![image](https://user-images.githubusercontent.com/4622125/141337227-c521293a-c040-41e3-aa08-2d56d7912077.png)

Manually modifying the form value to remove the string `"` characters fixes the validation issue and allows the deployment to begin:

![image](https://user-images.githubusercontent.com/4622125/141338513-bf2982c9-e477-40a7-bf0e-fce3f02d0d5d.png)

Odd behavior given this value is defaulted as an `array` type here:

https://github.com/Azure/missionlz/blob/e042e9d88234d5af01db094420cc0342d78e5403/src/bicep/mlz.bicep#L465

and read from this value, which is also an array:

https://github.com/Azure/missionlz/blob/e042e9d88234d5af01db094420cc0342d78e5403/src/bicep/mlz.bicep#L369-L373

will investigate and propose a fix

**Comment by brooke-hamilton** [comment]
For now we have removed the Azure portal templates from the `main` branch until we get a fix. The command-line deployments using the AZ CLI are not affected by this issue. See PR #516

**Comment by brooke-hamilton** [comment]
#482 is a similar issue and is being addressed too.

**Comment by brooke-hamilton** [comment]
This bug will be resolved with #482, which is currently in progress.
